src/features/collections/MatchPane.tsx

```diff
@@ -144,16 +146,15 @@ const CreateMatch = ({ collectionId }: { collectionId: string }) => {
   }, [matchers]);
 
   // Narrative selection
+  const [narrativeSearch, setNarrativeSearch] = useState('');
   const [narrativeSel, setNarrativeSel] = useState<SelectOption | undefined>();
-
-  const narrativeQuery = listWorkspaceInfo.useQuery({});
-
-  const narrativeOptions = (narrativeQuery?.data?.[0] || []).map((ws) => ({
-    value: ws[0],
-    label: ws[1],
-    data: ws,
+  const narrativeSearchParams = useParamsForNarrativeDropdown(narrativeSearch);
+  const narrativeQuery = getNarratives.useQuery(narrativeSearchParams);
+  const narrativeOptions = (narrativeQuery?.data?.hits || []).map((hit) => ({
+    value: [hit.access_group, hit.obj_id, hit.version].join('/'),
```

Comment:
Perhaps we should consider making a common function for formatting UPAs. I feel like I write it differently each time.